### DISCUSSION TITLE
Could this support MySQL?

### DISCUSSION THREAD

**Discussion by EmmanuelOga** [riverqueue/river#158] [Ideas]
Hey, I was wondering if there's any particular reason River uses Postgres.

AFAICT MySQL also supports SELECT FOR UPDATE and the MySQL driver for Go seems fast and well maitained:
https://github.com/go-sql-driver/mysql

Thanks

**Comment by bgentry** [comment]
Hi Emmanuel, at the moment River is quite Postgres-specific and is likely to stay that way for now. Our focus in the short term is to broaden the available audience of those already using Postgres with Go, to build out heavily requested features within the existing system, and to improve reliability and robustness. Off hand, a few other Postgres-specific things we use:

- `LISTEN` / `NOTIFY`
- `jsonb` and associated functions/operators
- Some `plpgsql` triggers

That's not to say it wouldn't be possible to make a similar style of queue that works with MySQL or that River might not someday support it, but for now it doesn't seem feasible for us. Also everything is currently fixed to using pgx v5 so just getting us to support other Postgres drivers is itself a sizable challenge.

**Reply by brandur** [reply]
+1. It'll be interesting to keep this in mind as we go. Another neat possibility is SQLite, which is about to get `jsonb`. Might be somewhat feasible to fallback to polling where `LISTEN`/`NOTIFY` aren't supported, and `json` where `jsonb` isn't supported.

For anyone else who happens to come across this thread: as Blake says, MySQL probably isn't coming soon, but please upvote @EmmanuelOga's top comment to express interest and this might help us prioritize down the line.

**Comment by EmmanuelOga** [comment]
Thanks for the quick answer!

**Comment by leaanthony** [comment]
Was this ever considered again?

**Reply by brandur** [reply]
@leaanthony It's definitely more plausible these days, and I've tried informal polls to gauge interest [1], but it's been hard to decide whether it'd be overall worth it or not — there seems to be some interest out there, but I wouldn't describe it anymore strongly than "some".

Do you want to tell us a little more about your use case?

[1] https://x.com/brandur/status/2082514592400335138
